src/Shared/ChangeWaves.cs

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Shared;
+using System;
+using System.Linq;
+
+namespace Microsoft.Build.Utilities
+{
+    internal enum ChangeWaveConversionState
+    {
+        NotConvertedYet,
+        Valid,
+        InvalidFormat,
+        OutOfRotation,
+        InvalidVersion
+    }
+
+    /// <summary>
+    /// All waves are enabled by default, meaning all features behind change wave versions are enabled.
+    /// </summary>
+    public class ChangeWaves
+    {
+        public static readonly string[] AllWaves = { Wave16_8, Wave16_10, Wave17_0 };
+        public static readonly Version[] AllWavesAsVersion = Array.ConvertAll<string, Version>(AllWaves, Version.Parse);
+        public const string Wave16_8 = "16.8";
+        public const string Wave16_10 = "16.10";
+        public const string Wave17_0 = "17.0";
+
+        /// <summary>
+        /// Special value indicating that all features behind change-waves should be enabled.
+        /// </summary>
+        public const string EnableAllFeatures = "999.999";
+
+        internal static readonly Version LowestWaveAsVersion = new Version(AllWaves[0]);
+        internal static readonly Version HighestWaveAsVersion = new Version(AllWaves[AllWaves.Length - 1]);
+        internal static readonly Version EnableAllFeaturesAsVersion = new Version(EnableAllFeatures);
+
+        internal static string LowestWave
+        {
+            get
+            {
+                return AllWaves[0];
+            }
+        }
+
+        internal static string HighestWave
+        {
+            get
+            {
+                return AllWaves[AllWaves.Length - 1];
```

Comment:
Apparently there's a new feature in C# (that you can't use yet) to say `AllWaves[~1]` here. That'll be nice.